packages/flutter/test/cupertino/scrollbar_test.dart

```diff
@@ -1060,6 +1060,50 @@ void main() {
     );
   });
 
+  testWidgets('Throw if interactive with the bar when no position attached', (WidgetTester tester) async {
+    final ScrollController scrollController = ScrollController();
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: MediaQuery(
+          data: const MediaQueryData(),
+          child: CupertinoScrollbar(
+            controller: scrollController,
+            thumbVisibility: true,
+            child: SingleChildScrollView(
+              controller: scrollController,
+              child: const SizedBox(
+                height: 1000.0,
+                width: 1000.0,
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    await tester.pumpAndSettle();
+
+    final ScrollPosition position = scrollController.position;
+    scrollController.detach(position);
+
+    final FlutterExceptionHandler? handler = FlutterError.onError;
+    FlutterErrorDetails? error;
+    FlutterError.onError = (FlutterErrorDetails details) {
+      error = details;
+    };
+
+    // long press the thumb
+    await tester.startGesture(const Offset(796.0, 50.0));
+    await tester.pump(kLongPressDuration);
+
+    expect(error, isNotNull);
```

Comment:
What is the error message here? The stack trace we got in the issue was not very helpful, does this help the user understand what is happening?

Reply:
I believe that the issue is an exception in production mode, I mean there should be assertion errors before this exception in debug mode.
I still modified some code because I wanted to be consistent with other places where getScrollbarDirection is used.
In fact, it's hard for test cases to test exactly what I've changed because the debug checking will fail the test first.

Reply:
I see. Sounds good, thank you.